pkg/internal/controller/controller.go

```diff
@@ -95,7 +95,10 @@ type watchDescription struct {
 }
 
 // Reconcile implements reconcile.Reconciler.
-func (c *Controller) Reconcile(ctx context.Context, req reconcile.Request) (reconcile.Result, error) {
+func (c *Controller) Reconcile(ctx context.Context, req reconcile.Request) (_ reconcile.Result, err error) {
```

Comment:
Sorry, what I meant by the other comment was:
1. Please add an option to `Controller` to make this configurable (i.E. `RecoverPanic bool`)
2. Make the handling here conditional (i.E. `if c.RecoverPanic { defer recover() }`)
3. Expose the option in `pkg/controller.Options`


The reason for favoring that over the current approach is that with the current approach we require ppl to read internal packages to understand how they can get panics recovered, which is not good UX.

Sorry for being unclear about this before.

Reply:
Understand, I will fix it. So what do you think of webhook handler? Should its panic default to be recovered or another option field to configure it?

Reply:
It should also have an option to set this up and by default panic

Reply:
Updated. It is hard to add the option for webhook. Maybe we can firstly add an option for controller reconcile.